docs/resource/README.md

```diff
@@ -99,10 +99,35 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
 <!-- semconv service_experimental -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
-| `service.instance.id` | string | The string ID of the service instance. [1] | `my-k8s-pod-deployment-1`; `627cc493-f310-47de-96bd-71410b7dec09` | Recommended |
+| `service.instance.id` | string | The string ID of the service instance. [1] | `627cc493-f310-47de-96bd-71410b7dec09` | Recommended |
 | `service.namespace` | string | A namespace for `service.name`. [2] | `Shop` | Recommended |
 
-**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations).
+**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words
+`service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to
+distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled
+service).
+
+Implementations are recommended to generate a random Version 1 or Version 4 [RFC
+4122](https://www.ietf.org/rfc/rfc4122.txt) UUID, but are free to use an inherent unique ID as the source of
+this value if stability is desirable. In that case, the ID SHOULD be used as source of a UUID Version 5 and
+SHOULD use the following UUID as the namespace: `4d63009a-8d0f-11ee-aad7-4c796ed8e320`.
+
+UUIDs are typically recommended, as only an opaque value for the purposes of identifying a service instance is
+needed. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file, the underlying
+data, such as pod name and namespace should be treated as confidential, being the user's choice to expose it
+or not via another resource attribute.
+
+For applications running behind an application server, such as unicorn, a stable identifier is not desirable,
+as each worker thread typically is seen as a different instance. For those cases, a UUID v1/v4 is always
+recommended.
+
+Collectors aren't recommended to set a `service.instance.id`: any identifying information that the Collector
```

Comment:
I'm not a huge fan of this paragraph.

E.g. in the collector we have a `mysql` receiver that can pull mysql database metrics.  For `service.instance.id` id DOES know an IP address it talks to, and could define an instance.id there.

I think usage of `service.instance.id` and the collector should NOT be restricted, and instead let users figure out best practices over time.

From a semocnv standpoint, we should just:

1. Define the *meaning* of attribute.
2. Outline a recommended way of generating the output.

Reply:
True, I had the case where the collector gets telemetry from an application via OTLP in mind but didn't consider scraping receivers. I still think it's useful to have some language here to avoid having a Collector implementation without taking into account the nuances we talked about here so extensively already. My major concern is the "two containers in a pod" use case, where both containers would have most of the same information but will typically lack a `container.id` or name. 

I gave it another shot. Let me know what you all think. If you still think it should be left out, I'll remove it.